### PR TITLE
#124 renames: Argument->Operand & Parameter->Argument

### PR DESCRIPTION
**Pull request by drewburlingame**
⚠️ for review only ⚠️ 

API changes

Marked Obsolete w/ redirect

- ArgumentMode.Parameter -> ArgumentMode.Operand
- ArgumentAttribute -> OperandAttribute
- ArgumentInfo.PropertyOrArgumentName -> ArgumentInfo.PropertyOrParameterName

- [x]  I still need to add tests to confirm these still work.  

Breaking

- CommandArgument -> CommandOperand
- ICommand.Arguments -> Operands
- IParameter -> IArgument
- Parsing.TokenType.Argument -> Operand

These are only available for advanced scenarios: custom help providers, invokers or argument transformations.  I think the impact will be minimal but still enough to warrant v3

### CODE REVIEW

File: CommandDotNet.Tests/FeatureTests/ApplicationMetadataTests.cs [hunk 1]
```diff
@@ -24,7 +24,7 @@ public void App_BasicHelp_DisplaysApplicationMetadata()
                 WhenArgs = "-h",
                 Then = { Result = @"app description
 
-Usage: SomeApp [options] [command]
+Usage: SomeApp [command] [options]
```
Comment (drewburlingame) [inline CommandDotNet.Tests/FeatureTests/ApplicationMetadataTests.cs]: turns out these have been out of order for a while.

File: CommandDotNet.Tests/FeatureTests/Arguments/AppsSettings_BoolMode.cs [hunk 2]
```diff
@@ -33,8 +33,8 @@ public AppsSettings_BoolMode(ITestOutputHelper output) : base(output)
   operand
 
 Options:
-  -h | --help  Show help information
-  --option"
+  --option
+  -h | --help  Show help information"
```
Comment (drewburlingame) [inline CommandDotNet.Tests/FeatureTests/Arguments/AppsSettings_BoolMode.cs]: took the opportunity to move help and version to the bottom of the list.  when i'm looking for help for a command, i'm already seeing it and i'm more interested in the help specific to the command so moving this down reduces the noise

File: CommandDotNet.Tests/FeatureTests/Arguments/DefaultArgumentMode.cs [hunk 3]
```diff
@@ -1,129 +1,221 @@
...
 
 namespace CommandDotNet.Tests.FeatureTests.Arguments
 {
-    public class DefaultArgumentMode : ScenarioTestBase<DefaultArgumentMode>
+    public class DefaultArgumentMode : TestBase
     {
-        private static readonly AppSettings OperandMode = TestAppSettings.BasicHelp.Clone(a => a.MethodArgumentMode = ArgumentMode.Parameter);
+        private static readonly AppSettings OperandMode = TestAppSettings.BasicHelp.Clone(a => a.MethodArgumentMode = ArgumentMode.Operand);
         private static readonly AppSettings OptionMode = TestAppSettings.BasicHelp.Clone(a => a.MethodArgumentMode = ArgumentMode.Option);
+        private static readonly AppSettings DeprecatedParameterMode = TestAppSettings.BasicHelp.Clone(a => a.MethodArgumentMode = ArgumentMode.Parameter);
```
Comment (drewburlingame) [inline CommandDotNet.Tests/FeatureTests/Arguments/DefaultArgumentMode.cs]: this test verifies the obsoleted `ArgumentMode.Parameter` and `ArgumentAttribute` still work.

Comment (drewburlingame) [reply CommandDotNet.Tests/FeatureTests/Arguments/DefaultArgumentMode.cs]: The rest of the changes will only impact people with: custom help providers, invokers or argument transformations

**Comment by drewburlingame** [comment]
@bilal-fazlani this is ready for review.  It's a lot of files.  To get the gist of it, review the [readme.md](https://github.com/bilal-fazlani/commanddotnet/pull/146/files#diff-04c6e90faac2675aa89e2176d2eec7d8) and the first 3 tests.

**Comment by bilal-fazlani** [comment]
Shouldn't we have merged this?

**Comment by drewburlingame** [comment]
It wasn't ready on it's own.  The changes are included in https://github.com/bilal-fazlani/commanddotnet/pull/150.

**Comment by bilal-fazlani** [comment]
Ahh ok 👍

**Pull request closed without merging**